pilot/pkg/config/coredatamodel/syntheticserviceentrycontroller.go

```diff
@@ -128,9 +124,19 @@ func (c *SyntheticServiceEntryController) HasSynced() bool {
 	return atomic.LoadUint32(&c.synced) != 0
 }
 
+func (c *SyntheticServiceEntryController) dispatch(config model.Config, event model.Event) {
+	if c.eventHandler != nil {
+		c.eventHandler(config, event)
+	}
+}
+
 // RegisterEventHandler registers a handler using the type as a key
 func (c *SyntheticServiceEntryController) RegisterEventHandler(typ string, handler func(model.Config, model.Event)) {
-	log.Warnf("registerEventHandler: %s", errUnsupported)
+	// TODO: investigate why it is called more than one
```

Comment:
fix it

Reply:
The reason it is being called twice is because once it is called in the MCP Discovery Constructor and the second time in config aggregate Register EventHandler. I think the best fix would be just the nil check above (only set it if it's nil), otherwise we would have to do a check for controller type (SyntheticServiceEntry type) in the [config aggregator](https://github.com/istio/istio/blob/4f0cb562a7128181519f0998d901e0e981010c78/pilot/pkg/config/aggregate/config.go#L177) which wouldn't be ideal.

Reply:
Then only `HandleCacheEvents` takes effect now, how do SSE updates trigger xds push?

Reply:
All the xds pushes are handled in the controller directly, we never relied on the eventHandlers for pushes.

Reply:
IC, it is called by `handleResponse`